plugins/catalog-import/src/api/AzureDevops.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2024 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { AzureIntegration } from '@backstage/integration';
+import { ScmAuthApi } from '@backstage/integration-react';
+import { ConfigApi } from '@backstage/core-plugin-api';
+import { getBranchName, getCatalogFilename } from '../components/helpers';
+import { createAzurePullRequest } from './AzureRepoApiClient';
+import parseGitUrl from 'git-url-parse';
+
+export interface AzureRepoParts {
+  tenantUrl: string;
+  repoName: string;
+  project: string;
+}
+
+export function parseAzureUrl(
+  repoUrl: string,
+  integration: AzureIntegration,
+): AzureRepoParts {
+  const { organization, owner, name } = parseGitUrl(repoUrl);
```

Comment:
We've had issues in the past using `git-url-parse` it doesn't play nicely with Azure DevOps Server as it can be hosted on a sub-path, this was the common pattern for TFS ( the previous name for Azure DevOps). This is what we have done to get around this: https://github.com/backstage/community-plugins/blob/a25418c4f52b759f78a1e4256b28f891e673f334/workspaces/azure-devops/plugins/azure-devops-backend/src/utils/azure-devops-utils.ts#L332

Reply:
There's a similar function here: https://github.com/backstage/backstage/blob/4eca697bf3c5a996795324444f605b918f4491be/packages/integration/src/azure/AzureUrl.ts#L25 

I'm not sure how usable that would be here though 🤔

Reply:
Interesting ... My expectation was that the `git-url-parser` would handle that on its own. I'm glad we caught this.
At first glance your first suggestion would work nicely but  from what I could tell it is part of a plugin.  What is the recommended approach here ? copy the function to the catalog-import  plugin ? or move that function to a common  package and reference it in both places ?  Adding a dependency to the plugin and using it doesn't feel right.

 I will pick this up when I return from vacation towards the end of the week.

Thank you !

Reply:
I  have applied the previously discussed changes 
  - Reverted the examples to default
  - Changed the  URL parser to use the one from azure-devops-backend. I did this by copying the function across.  Adding a dependency didn't feel right.
  
  **NOTE**: I tried to use `AzureUrl` class from `intergrations` but found out that it does not play nice with TFS servers.  The parsing logic will result to the owner being wrongly set to `tfs/organisation_name` rather than `organisation_name`
Might be worth considering to unify these in the integrations package

Looking forward to your feedback

Reply:
This looks great 👍